### PR TITLE
Add transChoice method in CRUDController

### PR DESCRIPTION
**Pull request by tifabien**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Added `CRUDController::transChoice` method
```

As transChoice is deprecated in admin class it is useful to have a transChoice method in CRUDController. 
Note that I used scalar type and return type declarations as it seems we are now only compatible with PHP 7.

### CODE REVIEW

**Comment by OskarStark** [comment]
> Note that I used scalar type and return type declarations as it seems we are now only compatible with PHP 7.

👍

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -1530,6 +1530,22 @@ protected function preList(Request $request)
...
+    /**
+     * @param string      $id
+     * @param int         $number
+     * @param array       $parameters
+     * @param string|null $domain
+     * @param string|null $locale
+     *
+     * @return string
+     */
+    final protected function transChoice(string $id, int $number, array $parameters = [], string $domain = null, string $locale = null): string
```
[the remark this answers is not in the document]
Comment (tifabien) [reply src/Controller/CRUDController.php]: Indeed it seems working correctly even when no `$locale` param passed!

File: src/Controller/CRUDController.php [hunk 2]
```diff
@@ -1530,6 +1530,13 @@ protected function preList(Request $request)
         return $this->get('translator')->trans($id, $parameters, $domain, $locale);
     }
 
+    final protected function transChoice(string $id, int $number, array $parameters = [], ?string $domain = null, ?string $locale = null): string
```
Comment (greg0ire) [inline src/Controller/CRUDController.php]:
```suggestion
    final protected function transChoice(
        string $id,
        int $number,
        array $parameters = [],
        ?string $domain = null,
        ?string $locale = null
    ): string
```

**Comment by greg0ire** [comment]
Uuuuuh, why would you need a `transChoice` method in a controller? Controllers should be exclusively about HTTP.

**Comment by jvasseur** [comment]
In addition to what @greg0ire said, `transChoice` is deprecated since Symfony 4.2 so I don't think we should add a shortcut method for it.

**Comment by tifabien** [comment]
I didn't know they deprecated `transChoice` method... So how could we manage a `transChoice` usage in an extended CRUDController (E.g. add a flashMessage)  because `transChoice` is also deprecated in the `AbstractAdmin` class? 

> Controllers should be exclusively about HTTP.

We could maybe deprecates the `trans` method?

**Comment by greg0ire** [comment]
> We could maybe deprecates the trans method?

Yeah, probably!

**Pull request closed without merging**